extra_foam/utils.py

```diff
@@ -42,6 +42,31 @@ def timed_f(*args, **kwargs):
     return wrap
 
 
+
+class BlockTimer:
+    """A context manager for measuring the execution time of a code block
+
+    For example::
+
+        >>> with BlockTimer("foo"):
+        ...     time.sleep(1)
+        ...
+        Execution of foo: 1.001s
+    """
+    def __init__(self, label="block", enabled=True):
+        self._label = label
+        self._enabled = enabled
+
+    def __enter__(self):
+        self.start = time.perf_counter()
+        return self
+
+    def __exit__(self, *_):
+        duration = time.perf_counter() - self.start
+        if self._enabled:
+            print(f"Execution of {self._label}: {duration:.4f}s")
```

Comment:
`logger.info`?

Reply:
Oooo yes that's better, fixed in 175daf75.

Reply:
Good catch! I had the impression that this is an internal timer. I agree that it is better to use `logger.info` if we'd like to expose it outside. We can also use `logger.debug` in the `profiler`.